public/app/plugins/datasource/elasticsearch/datasource.js

```diff
@@ -47,10 +47,19 @@ function (angular, _, moment, kbn, ElasticQueryBuilder, IndexPattern, ElasticRes
     };
 
     this._get = function(url) {
-      return this._request('GET', this.indexPattern.getIndexForToday() + url).then(function(results) {
-        results.data.$$config = results.config;
-        return results.data;
-      });
+      var range = timeSrv.timeRange();
+      var index_list = this.indexPattern.getIndexList(range.from.valueOf(), range.to.valueOf());
```

Comment:
I think, autocomplete may not work properly without this change. For example, you haven't index for today, but just want to look at data for previous week. In this case `getIndexForToday()` returns not existed index name.